### PR TITLE
Update Bazel to 5.3.0

### PR DESCRIPTION
**Pull request by aherrmann**
The previously used Bazel version 4.2.1 is incompatible with the now by default installed Visual Studio 2022 on the Github actions runner.
This prevented the use of a C++ binaries in test cases in https://github.com/tweag/rules_sh/pull/34.

### CODE REVIEW

File: sh/repositories.bzl [hunk 1]
```diff
@@ -27,18 +27,20 @@ def rules_sh_dependencies():
     maybe(
         http_archive,
         name = "bazel_skylib",
-        sha256 = "f7be3474d42aae265405a592bb7da8e171919d74c16f082a5457840f06054728",
+        sha256 = "74d544d96f4a5bb630d465ca8bbcfe231e3594e5aae57e1edbf17a6eb3ca2506",
         urls = [
-            "https://mirror.bazel.build/github.com/bazelbuild/bazel-skylib/releases/download/1.2.1/bazel-skylib-1.2.1.tar.gz",
-            "https://github.com/bazelbuild/bazel-skylib/releases/download/1.2.1/bazel-skylib-1.2.1.tar.gz",
+            "https://mirror.bazel.build/github.com/bazelbuild/bazel-skylib/releases/download/1.3.0/bazel-skylib-1.3.0.tar.gz",
+            "https://github.com/bazelbuild/bazel-skylib/releases/download/1.3.0/bazel-skylib-1.3.0.tar.gz",
```
Comment (avdv) [inline sh/repositories.bzl:33]: Should we also update the dependency in the `MODULE.bazel`?

File: sh/repositories.bzl [hunk 2]
```diff
@@ -27,18 +27,20 @@ def rules_sh_dependencies():
...
-        "platforms",
-        sha256 = "23566db029006fe23d8140d14514ada8c742d82b51973b4d331ee423c75a0bfa",
-        strip_prefix = "platforms-46993efdd33b73649796c5fc5c9efb193ae19d51",
-        urls = ["https://github.com/bazelbuild/platforms/archive/46993efdd33b73649796c5fc5c9efb193ae19d51.tar.gz"],
+        name = "platforms",
+        sha256 = "5308fc1d8865406a49427ba24a9ab53087f17f5266a7aabbfc28823f3916e1ca",
+        urls = [
+            "https://mirror.bazel.build/github.com/bazelbuild/platforms/releases/download/0.0.6/platforms-0.0.6.tar.gz",
+            "https://github.com/bazelbuild/platforms/releases/download/0.0.6/platforms-0.0.6.tar.gz",
+        ],
```
Comment (avdv) [inline sh/repositories.bzl:43]: Likewise.

**Review by avdv — CHANGES_REQUESTED** [review]

File: sh/repositories.bzl (hunk 1 above)
Comment (aherrmann) [reply sh/repositories.bzl:33]: Good catch! I forgot about that. Fixed.

Comment (aherrmann) [reply sh/repositories.bzl:33]: Ah, not so fast. The very latest versions are not available in the BCR, yet. I've bumped to the most recent available.

**Review by avdv — APPROVED** [review]

**Pull request merged**